File: src/scala/Programme.scala
/******************************************************************************

@LICENCE@
 
 *****************************************************************************/
 
package buyat.api.entities

/**
 * Represents a buy.at programme.
 * @author buy.at
 * @version 1.0
 * @since 1.0
 * @param _programmeId Programme ID.
 * @param _programmeName Programme name.
 * @param _programmeURL Programme URL.
 * @param _hasFeed Whether or not the programme has at least one feed.
 */
class Programme(_programmeId:Int, _programmeName:String, _programmeURL:String, _hasFeed:Boolean) extends APIEntity {

  /**
   * Unique ID of the programme.
   */
  var programmeId:Int = _programmeId

  /**
   * Name of the programme.
   */
  var programmeName:String = _programmeName

  /**
   * Tracking URL for the programme.
   */
  var programmeURL:String = _programmeURL

  /**
   * Whether or not this programme has feeds.
   */
  var hasFeed:Boolean = _hasFeed
}

File: src/csharp/Programme.cs
/******************************************************************************

@LICENCE@
 
 *****************************************************************************/

using System;
using System.Collections.Generic;
using System.Text;

namespace BuyAtAPI
{

    public class Programme
    {

        private BuyatAffiliateEntitiesProgramme soapProgramme;

        /// <summary>
        /// Constructor, wrapping SOAP object
        /// </summary>
        public Programme(BuyatAffiliateEntitiesProgramme soapProgramme)
        {
            this.soapProgramme = soapProgramme;
        }

        /// <summary>
        /// The programme ID
        /// </summary>
        public int ProgrammeID
        {
            get
            {
                try
                {
                    return Convert.ToInt32(soapProgramme.programme.programme_id);
                }
                catch (FormatException)
                {
                    return -1;
                }
            }
        }

        /// <summary>
        /// The programme name
        /// </summary>
        public string ProgrammeName
        {
            get
            {
                return soapProgramme.programme.programme_name;
            }
        }

        /// <summary>
        /// The programme URL
        /// </summary>
        public string ProgrammeURL
        {
            get
            {
                return soapProgramme.programme.programme_url;
            }
        }

        /// <summary>
        /// Whether the programme has feeds associated with it
        /// </summary>
        public bool HasFeed
        {
            get
            {
                return soapProgramme.programme.has_feed;
            }
        }
    }
}
